amazon/src/hooks/input/models/InputAction.ts

```diff
@@ -0,0 +1,8 @@
+export const INPUT_ACTION_CHANGE = 'CHANGE';
+export const INPUT_ACTION_BLUR = 'BLUR';
+export const INPUT_ACTION_CLEAR = 'CLEAR';
+
+export type InputActionType =
+  | typeof INPUT_ACTION_CHANGE
+  | typeof INPUT_ACTION_BLUR
+  | typeof INPUT_ACTION_CLEAR;
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using enum for input actions</b></div>
<div id="fix">

Consider using an enum instead of separate constants for input action types. This could improve type safety and make the code more maintainable.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,8 +1,7 @@
 -export const INPUT_ACTION_CHANGE = 'CHANGE';
 -export const INPUT_ACTION_BLUR = 'BLUR';
 -export const INPUT_ACTION_CLEAR = 'CLEAR';
 -
 -export type InputActionType =
 -  | typeof INPUT_ACTION_CHANGE
 -  | typeof INPUT_ACTION_BLUR
 -  | typeof INPUT_ACTION_CLEAR;
 +export enum InputActionType {
 +  CHANGE = 'CHANGE',
 +  BLUR = 'BLUR',
 +  CLEAR = 'CLEAR'
 +}

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/amazon-clone/pull/4#issuecomment-2466676739>#545432</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged